### ISSUE TITLE
vcdextract fails to locate CDDA for a specific image

### ISSUE THREAD

**Issue by mistydemeo** [cyberwarriorx/vcdextract#11]
When extracting a specific BIN/CUE image, vcdextract fails to grab the CDDA and crashes when trying to dereference a null pointer. Looks like it's happening when hitting the first CDDA track (track 3 on the disc). Specifically here: https://github.com/cyberwarriorx/vcdextract/blob/master/src/ISOExtract.cpp#L561

Where `FADToTrack` is returning a null pointer. Here's what I'm seeing for the values used by the lookup: `dirrec[i].LocationOfExtentL: 244897; cdinfo.trackinfo[trackindex].fileoffset: 0; total: 245047`

```
FILE "Lunar Silver Star Story Complete MPEG (JAP) (DW0559).bin" BINARY
  TRACK 01 MODE1/2352
    INDEX 01 00:00:00
  TRACK 02 MODE2/2352
    PREGAP 00:03:00
    INDEX 01 07:55:41
  TRACK 03 AUDIO
    PREGAP 00:02:00
    INDEX 01 54:20:22
  TRACK 04 AUDIO
    INDEX 00 56:44:70
    INDEX 01 56:46:69
```

**Comment by mistydemeo** [comment]
Digging in more on this same disc image, I'm also seeing it fetching incorrect FAD end/FAD start data, causing it to hit an endless loop when reading track 4:

```
cdinfo.trackinfo[i].fadend: 525
cdinfo.trackinfo[i].fadstart: 256044
num_sectors: -255518
```

**Comment by cyberwarriorx** [comment]
I don't have an exact copy of your image, however mine looks similar:

```
FILE "LUNAR_SILVER_STAR_STORY_MPEG.BIN" BINARY
  TRACK 01 MODE1/2352
    INDEX 01 00:00:00
  TRACK 02 MODE2/2352
    PREGAP 00:03:00
    INDEX 01 07:55:41
  TRACK 03 AUDIO
    PREGAP 00:02:00
    INDEX 01 54:20:22
  TRACK 04 AUDIO
    INDEX 00 56:44:70
    INDEX 01 56:46:69
  REM LEAD-OUT 57:00:41

```

I'm seeing an issue when trying to read the image size for the last track fad end calculation. It's looking for the image file in the CWD of the program instead of the specified cue/bin directory. Not sure if that's what you're seeing too. Should be a simple enough fix.

**Comment by cyberwarriorx** [comment]
Fixed the issue above in 617e7156911a731c8e8f49e9561f9fb263563d74, let me know if that fixes your originally reported problem.
